hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotFileCache.java

```diff
@@ -251,6 +260,25 @@ private void refreshCache() throws IOException {
     this.snapshots.putAll(newSnapshots);
   }
 
+  @VisibleForTesting
+  List<String> getSnapshotsInProgress() throws IOException {
+    List<String> snapshotInProgress = Lists.newArrayList();
+    // only add those files to the cache, but not to the known snapshots
+    FileStatus[] snapshotsInProgress = CommonFSUtils.listStatus(fs,
+      new Path(snapshotDir, SnapshotDescriptionUtils.SNAPSHOT_TMP_DIR_NAME));
+
+    if (!ArrayUtils.isEmpty(snapshotsInProgress)) {
+      for (FileStatus snapshot : snapshotsInProgress) {
+        try {
+          snapshotInProgress.addAll(fileInspector.filesUnderSnapshot(snapshot.getPath()));
+        } catch (CorruptedSnapshotException cse) {
+          LOG.debug("Corrupted in-progress snapshot file exception, ignored.", cse);
```

Comment:
Could this be handled better? It seems that in the event of an exception, the entirety of the in-progress snapshot would excluded.

Logging the exception doesn't tell us anything useful because the `SnapshotDescription` isn't party to the message. Maybe we should log the snapshot name and table name. At WARN level?

Reply:
It is hard to say. The CorruptedSnapshotException will only be thrown out when it reads a corrupted .snapshotinfo file. There are two cases, one is normal, which .snapshotinfo is being written and it reads incomplete one; the other is that it is a real corrupted .snapshotinfo. The original code assumes the first case (maybe in reality, there are more first cases than the second one). Change it to info level? The exception has the following info
```
throw new CorruptedSnapshotException("Couldn't read snapshot info from:" + snapshotInfo, e);
```
It is the path of .snapshotinfo, it has snapshot name embedded in the path (does not have table name). Let me see if table name can be added.

Reply:
It will be hard to put table name here. For an example, in the context of ExportSnapshot job, at the destination cluster, the .snapshotinfo is written to tmp directory first. There is no table existing yet, table may be cloned from snapshot after ExportSnapshot job is done.

Reply:
`CorruptedSnapshotException` contains a copy of `SnapshotDescription`. Where this is logged, instead of logging the exception, it can grab the snaps tho info out of the description field to build a more meaningful log message.

I don't know about the log level. I guess it depends on if you think this is a really severe problem or not. If not, `debug` is fine. If it's the only sign an operator has of a bad problem, or it should be `info` or `warn`.

Reply:
yeah, it has multiple constructors, in this specific case, there is no SnapshotDescription provided (reading SnapshotDescription results into IOException due to corrupted/missing files). I changed log level from debug to info